### DISCUSSION TITLE
Document how to get better debugging information

### DISCUSSION THREAD

**Discussion by kontur** [websideproject/paddle-billing-client#214] [Ideas]
I was implementing a basic transaction setup first time with this library. A silly typo in the parameter code was very hard to debug, since the library hides any actual `errors` details returned from the API and only returns endpoint and error code.

After some digging I found `apiclient.request_strategies.Request_Strategy._handle_bad_response` which uses the `error_handler` and saw `paddle_billing_client.client.PaddleApiClient` can be initiated with a custom `error_handler` argument that can be customized.

It might be worth documenting this as example, or even for sandbox mode enable a more detailed error handler like this by default:

```
class CustomErrorHandler(BaseErrorHandler):
    @staticmethod
    def get_exception(response: Response) -> exceptions.APIRequestError:
        status_code = response.get_status_code()
        exception_class = exceptions.UnexpectedError

        if 300 <= status_code < 400:
            exception_class = exceptions.RedirectionError
        elif 400 <= status_code < 500:
            exception_class = exceptions.ClientError
        elif 500 <= status_code < 600:
            exception_class = exceptions.ServerError

        return exception_class(
            message=(
                f"{status_code} Error: {response.get_status_reason()} "
                f"for url: {response.get_requested_url()}"
                f"with raw data: {response.get_raw_data()}" # <<<<<<<<<<<<<< this is the informative bit
            ),
            status_code=status_code,
            info=response.get_raw_data(),
        )
...
client = PaddleApiClient(
        base_url="https://sandbox-api.paddle.com", 
        authentication_method=HeaderAuthentication(token=...),
        error_handler=CustomErrorHandler
    )
```

**Comment by bgervan** [comment]
I will check it, thanks. I was having the same issues with the debug, but was not painful enough to look into it yet.

**Reply by bgervan** [reply]
Created a new release which includes that and Readme usage doc. 
https://github.com/websideproject/paddle-billing-client/releases/tag/v0.2.3

I didn't add as default, as it's easy to customize based on the example

**Comment by kontur** [comment]
Super, nice, the API returns very helpful pointers for `400` errors, this makes it a lot more easy to spot mistakes in the signature of the call 👍
